azurelinuxagent/ga/update.py

```diff
@@ -267,6 +268,24 @@ def run(self, debug=False): # pylint: disable=R0912,R0914
             logger.info(os_info_msg)
             add_event(AGENT_NAME, op=WALAEventOperation.OSInfo, message=os_info_msg)
 
+            #
+            # Perform initialization tasks
+            #
+            from azurelinuxagent.ga.exthandlers import get_exthandlers_handler, migrate_handler_state
```

Comment:
Should this import be up top of the file?

Reply:
ya; i tried to fix that and it ends up breaking up the tests... we have several of those problematic imports across the code base... yet another cleanup task to do

Reply:
Hm. What were the errors the tests were having? Should we document this?

Reply:
you can try moving it in the current code and check the errors... i don't think we need to document it, all of us are aware of this (now you are :))